Userland/Libraries/LibWeb/FileAPI/File.cpp

```diff
@@ -77,4 +87,53 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<File>> File::construct_impl(JS::Realm& real
     return create(realm, file_bits, file_name, options);
 }
 
+WebIDL::ExceptionOr<void> File::serialization_steps(HTML::SerializationRecord& record, bool)
+{
+    auto& vm = this->vm();
+
+    record.append(HTML::ValueTag::FileObject);
+
+    // FIXME: 1. Set serialized.[[SnapshotState]] to value’s snapshot state.
+
+    // NON-STANDARD: FileAPI spec doesn't specify that type should be serialized, although
+    //               to be conformant with other browsers this needs to be serialized.
+    TRY(HTML::serialize_string(vm, record, m_type));
+
+    // 2. Set serialized.[[ByteSequence]] to value’s underlying byte sequence.
+    TRY(HTML::serialize_bytes(vm, record, m_byte_buffer.bytes()));
+
+    // 3. Set serialized.[[Name]] to the value of value’s name attribute.
+    TRY(HTML::serialize_string(vm, record, m_name));
+
+    // 4. Set serialized.[[LastModified]] to the value of value’s lastModified attribute.
+    record.append(bit_cast<u32*>(&m_last_modified), 2);
+
+    return {};
+}
+
+WebIDL::ExceptionOr<void> File::deserialization_steps(ReadonlySpan<u32> const& record, size_t& position)
+{
+    auto& vm = this->vm();
+
+    // FIXME: 1. Set value’s snapshot state to serialized.[[SnapshotState]].
+
+    // NON-STANDARD: FileAPI spec doesn't specify that type should be deserialized, although
+    //               to be conformant with other browsers this needs to be deserialized.
+    m_type = TRY(HTML::deserialize_string(vm, record, position));
+
+    // 2. Set value’s underlying byte sequence to serialized.[[ByteSequence]].
+    m_byte_buffer = TRY(HTML::deserialize_bytes(vm, record, position));
+
+    // 3. Initialize the value of value’s name attribute to serialized.[[Name]].
+    m_name = TRY(HTML::deserialize_string(vm, record, position));
+
+    // 4. Initialize the value of value’s lastModified attribute to serialized.[[LastModified]].
+    u32 bits[2] = {};
+    bits[0] = record[position++];
+    bits[1] = record[position++];
+    m_last_modified = *bit_cast<i64*>(&bits);
```

Comment:
This looks kind of dangerous. do the other helpers have any checks that there's actually N bytes 'left' in record to read from before doing `blah = record[position++]` ?

Reply:
Unfortuntately we assume that there are N bytes left in most of our deserialization implementation. Shouldn't be too hard to fix up later, we should though agree on what kind of error and message we should throw back to the JS caller, if any.

Reply:
I am leaving this as is for now since we have this pattern going on in quite a few places in StructuredSerialize